pyproject.toml

```diff
@@ -10,6 +10,151 @@ requires = [
   "oldest-supported-numpy",
 ]
 
+[project]
+name = "yt"
+version = "4.2.dev0"
+description = "An analysis and visualization toolkit for volumetric data"
+authors = [
+    { name = "The yt project", email = "yt-dev@python.org" },
+]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Environment :: Console",
+    "Framework :: Matplotlib",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: MacOS :: MacOS X",
+    "Operating System :: POSIX :: AIX",
+    "Operating System :: POSIX :: Linux",
+    "Programming Language :: C",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Topic :: Scientific/Engineering :: Astronomy",
+    "Topic :: Scientific/Engineering :: Physics",
+    "Topic :: Scientific/Engineering :: Visualization",
+]
+keywords = [
+    "astronomy astrophysics visualization amr adaptivemeshrefinement",
+]
+requires-python = ">=3.8"
+dependencies = [
+    "cmyt>=1.1.2",
+    "ipywidgets>=8.0.0",
+    "matplotlib!=3.4.2,>=3.2",
+    "more-itertools>=8.4",
+    "numpy>=1.17.5",
+    "packaging>=20.9",
+    "pillow>=6.2.1",
+    "pyparsing>=2.0.3",
+    "tomli-w>=0.4.0",
+    "tqdm>=3.4.0",
+    "unyt>=2.9.2,<3.0",
+    "tomli>=1.2.3;python_version < '3.11'",
+]
+
+[project.readme]
+file = "README.md"
+content-type = "text/markdown"
+
+[project.license]
+text = "BSD 3-Clause"
+
+[project.urls]
+Homepage = "https://yt-project.org/"
+Documentation = "https://yt-project.org/doc/"
+Source = "https://github.com/yt-project/yt/"
+Tracker = "https://github.com/yt-project/yt/issues"
+
+[project.entry-points."nose.plugins.0.10"]
+answer-testing = "yt.utilities.answer_testing.framework:AnswerTesting"
```

Comment:
it's not clear yet that this auto-migration will work. I'm hoping CI doesn't rely on it because it needs to be dropped or migrated to pytest eventually. If nothing break in the first run I'll try again with this section completely removed to see how things go.

Reply:
Looks like it's used only on GHA for answer tests, so I'll keep it in this PR but it can be removed when #4122 goes in